src/app/services/xml-parsers/header-parser.ts

```diff
@@ -0,0 +1,218 @@
+      editor: queryAndParseElements<GenericElement>(xml, 'editor', this.genericElemParser),
+      biblScope: queryAndParseElements<GenericElement>(xml, 'biblScope', this.genericElemParser),
+    };
+  }
+}
+
+export class NotesStmtParser extends GenericParser implements Parser<XMLElement> {
+  private notesStmt = createParser(NoteParser, this.genericParse);
+
+  parse(xml: XMLElement): NotesStmt {
+    return {
+      type: NotesStmt,
+      class: getClass(xml),
+      content: parseChildren(xml, this.genericParse, true),
+      attributes: this.attributeParser.parse(xml),
+      notes: queryAndParseElements<Note>(xml, 'note', this.notesStmt),
+      relatedItems: queryAndParseElements<GenericElement>(xml, 'relatedItem', this.genericElemParser),
+    };
+  }
+}
+
+export class SourceDescParser extends GenericParser implements Parser<XMLElement> {
+  private msDescParser = createParser(MsDescParser, this.genericParse);
+
+  parse(xml: XMLElement): SourceDesc {
+    return {
+      type: SourceDesc,
+      class: getClass(xml),
+      content: parseChildren(xml, this.genericParse, true),
+      attributes: this.attributeParser.parse(xml),
+      structuredData: Array.from(xml.children).filter(el => el.tagName === 'p').length !== xml.children.length,
+      msDesc: queryAndParseElement<MsDesc>(xml, 'note', this.msDescParser),
+      bibl: queryAndParseElements<GenericElement>(xml, 'bibl', this.genericElemParser),
+      biblFull: queryAndParseElements<GenericElement>(xml, 'biblFull', this.genericElemParser),
+      biblStruct: queryAndParseElements<GenericElement>(xml, 'biblStruct', this.genericElemParser),
+      recordingStmt: queryAndParseElements<GenericElement>(xml, 'recordingStmt', this.genericElemParser),
+      scriptStmt: queryAndParseElements<GenericElement>(xml, 'scriptStmt', this.genericElemParser),
+    };
+  }
+}
+
+export class ExtentParser extends AttrParser implements Parser<XMLElement> {
+  parse(xml: XMLElement): Extent {
+    return {
+      type: Object,
+      class: getClass(xml),
+      content: parseChildren(xml, this.genericParse),
+      attributes: this.attributeParser.parse(xml),
+    };
+  }
+}
+
+export class FileDescParser extends AttrParser implements Parser<XMLElement> {
+  private excludeFromParsing = [
+    'listBibl',
+    'listEvent',
+    'listOrg',
+    'listPerson',
+    'listPlace',
+    'listWit',
+    'sourceDesc list',
+  ];
+  private titleStmtParser = createParser(TitleStmtParser, this.genericParse);
+  private editionStmtParser = createParser(EditionStmtParser, this.genericParse);
+  private publicationStmtParser = createParser(PublicationStmtParser, this.genericParse);
+  private seriesStmtParser = createParser(SeriesStmtParser, this.genericParse);
+  private notesStmtParser = createParser(NotesStmtParser, this.genericParse);
+  private sourceDescParser = createParser(SourceDescParser, this.genericParse);
+  private extentParser = createParser(ExtentParser, this.genericParse);
```

Comment:
Since you use these only in a single point, you can create them on the fly when you pass the parser parameters to the `queryAndParseElement` function